keras_cv/layers/preprocessing/random_shear.py

```diff
@@ -148,3 +156,185 @@ def get_config(self):
             }
         )
         return config
+
+    def _augment(self, inputs):
+        image = inputs.get("images", None)
+        label = inputs.get("labels", None)
+        bounding_box = inputs.get("bounding_boxes", None)
+        transformation = self.get_random_transformation(
+            image=image, label=label, bounding_box=bounding_box
+        )  # pylint: disable=assignment-from-none
+        image = self.augment_image(image, transformation=transformation)
+        result = {"images": image}
+        if label is not None:
+            label = self.augment_label(label, transformation=transformation)
+            result["labels"] = label
+        if bounding_box is not None:
+            # changed augment_bounding_box requires image as well
+            # for unnormalizing bbox coordinates
+            bounding_box = self.augment_bounding_box(
+                image, bounding_box, transformation=transformation
+            )
+            result["bounding_boxes"] = bounding_box
+        return result
+
+    def augment_bounding_box(self, image, bounding_boxes, transformation):
+        """args: image : takes a single image H,W,C,
+        bounding_boxes: take bbox coordinates [N,4] -> [y1,x1,y2,x2]
+        transformation: takes tuple for x,y transformation None if no transformation"""
+        height, width, _ = image.shape
+        image = tf.expand_dims(image, axis=0)
+        if self.bounding_box_format != "xyxy":
+            bounding_boxes = keras_cv.bounding_box.convert_format(
+                bounding_boxes,
+                source=self.bounding_box_format,
+                target="xyxy",
+                images=image,
+            )
+        x, y = transformation
+        # apply horizontal shear
+        if x is not None:
+            bounding_boxes = self._apply_horizontal_transformation_to_bounding_box(
+                bounding_boxes, x
+            )
+        # apply vertical shear
+        if y is not None:
+            bounding_boxes = self._apply_vertical_transformation_to_bounding_box(
+                bounding_boxes, y
+            )
+        # clip bounding boxes value to 0-image height and 0-image width
+        bounding_boxes = RandomShear.clip_bounding_box(height, width, bounding_boxes)
+        if self.bounding_box_format != "xyxy":
+            bounding_boxes = keras_cv.bounding_box.convert_format(
+                bounding_boxes,
+                source="xyxy",
+                target=self.bounding_box_format,
+                images=image,
+            )
+        return bounding_boxes
```

Comment:
you should specify the bounding_box dtype in convert_format or cast your output back to original bounding box dtype.